Test/ContentPipeline/TextureProcessorTests.cs

```diff
@@ -115,6 +115,100 @@ public void Mipmap()
         }
 
         [Test]
+        public void MipmapNonSquarePowerOfTwo()
+        {
+            var context = new TestProcessorContext(TargetPlatform.Windows, "dummy.xnb");
+
+            var processor = new TextureProcessor
+            {
+                ColorKeyEnabled = false,
+                GenerateMipmaps = true,
+                PremultiplyAlpha = false,
+                ResizeToPowerOfTwo = false,
+                TextureFormat = TextureProcessorOutputFormat.Color
+            };
+
+            var face = new PixelBitmapContent<Color>(16, 8);
+            Fill(face, Color.Red);
+            var input = new Texture2DContent();
+            input.Faces[0] = face;
+
+            var output = processor.Process(input, context);
+
+            Assert.NotNull(output);
+            Assert.AreEqual(1, output.Faces.Count);
+
+            var outChain = output.Faces[0];
+            Assert.AreEqual(5, outChain.Count);
+
+            var width = 16;
+            var height = 8;
+
+            foreach (var outFace in outChain)
+            {
+                Assert.AreEqual(width, outFace.Width);
+                Assert.AreEqual(height, outFace.Height);
+
+                var bitmap = (PixelBitmapContent<Color>)outFace;
+                for (var y = 0; y < height; y++)
+                    for (var x = 0; x < width; x++)
+                        Assert.AreEqual(Color.Red, bitmap.GetPixel(x, y));
+
+                if (width > 1)
+                    width /= 2;
+                if (height > 1)
+                    height /= 2;
+            }
+        }
+
+        [Test]
+        public void MipmapNonSquareNonPowerOfTwo()
+        {
+            var context = new TestProcessorContext(TargetPlatform.Windows, "dummy.xnb");
+
+            var processor = new TextureProcessor
+            {
+                ColorKeyEnabled = false,
+                GenerateMipmaps = true,
+                PremultiplyAlpha = false,
+                ResizeToPowerOfTwo = false,
+                TextureFormat = TextureProcessorOutputFormat.Color
+            };
+
+            var face = new PixelBitmapContent<Color>(23, 5);
+            Fill(face, Color.Red);
+            var input = new Texture2DContent();
+            input.Faces[0] = face;
+
+            var output = processor.Process(input, context);
+
+            Assert.NotNull(output);
+            Assert.AreEqual(1, output.Faces.Count);
+
+            var outChain = output.Faces[0];
+            Assert.AreEqual(5, outChain.Count);
+
+            var width = 23;
+            var height = 5;
```

Comment:
Maybe use these to initialize the `PixelBitmapContent` above to avoid a mistake later.